website/pages/blog/building-cloudquery.md

```diff
@@ -0,0 +1,102 @@
+
+<BlogHeader/>
+
+Data Integration is a 30+ year-old problem of extracting data from APIs, normalizing it and loading it into various different destinations such as databases, data lakes and files. In the last 20 years the problem has become acute, especially in the marketing and business analytics space, but recently also in the infrastructure world, with the explosion of cloud vendors and infrastructure services.
+
+Even though data integration is an old problem and will never be [completely](https://twitter.com/mattrickard/status/1542193426979909634) solved (at least while SaaS services don’t get better at having consistent machine readable APIs), we believe there is a big opportunity to address this problem at scale and achieve a new local optimum, and it can be done right now.
+
+Addressing this problem comes with significant challenges. In this post, I’ll go over these challenges, why they’re hard, and the solutions we landed on in CloudQuery’s design.
+
+# Engineering Requirements and Challenges
+
+Before we go over the technical and architecture decisions we made in CloudQuery, let’s cover the main challenges and requirements of an ELT framework:
+
+- **API Coverage**: Scaling and supporting the number of APIs covered in an efficient manner (i.e less developers, more APIs supported)--leveraging the community is key.
+- **Scaling Destinations**: Unlike sources, destinations are finite but the right architecture should ensure source and destinations are decoupled correctly so new destinations will work out-of-the-box and won't require any changes in sources (otherwise development work will grow exponentially).
+- **Performance**: Running ELT workload is compute intensive and can get expensive depending on the number of APIs to extract from. Having high performance and concurrent architecture is key to both making cost as low as possible and bringing information as fast as possible from sources to destinations.
+
+# Technical Deep Dive
+
+This section will discuss the most important decisions and solutions we landed on, organized by the challenges listed above.
+
+## API Coverage
+
+### Open Source
+
+There is no vendor that can support all integrations in-house thus the best and only way in our opinion is to go the open-source way and give the ability for others to contribute to our official plugins and develop community plugins without being blocked by a single vendor. Good recent examples that dealt with the “infinite APIs integrations” problem and took the open source route successfully are Terraform and Pulumi.
+
+### Pluggable Sources
+
+Pluggable sources are key to scale API coverage both from development and usability perspective and solve the following challenges:
+
+- **Binary Size:** Users download only the plugins they want to use to ensure good performance and user experience without a binary that grows infinitely and will eventually cause not only binary size issues but library clashes from different services.
```

Comment:
```suggestion
- **Binary Size**: Users download only the plugins they want to use to ensure good performance and user experience without a binary that grows infinitely and will eventually cause not only binary size issues but library clashes from different services.
```
Or the other way around (update the 2 items below and move `:` inside the **strong** section)

Reply:
Done